registry/quilt_server/views.py

```diff
@@ -2337,3 +2475,35 @@ def comments_list(owner, package_name):
     comments = Comment.query.filter_by(package=package).order_by(Comment.created)
 
     return dict(comments=map(_comment_dict, comments))
+
+@app.route('/api/auth/get_credentials', methods=['GET'])
+@api(require_login=True)
+@as_json
+def get_credentials():
+    """
+    Obtains credentials corresponding to your role.
+
+    Returns a JSON object with three keys:
+        AccessKeyId(string): access key ID
+        SecretKey(string): secret key
+        SessionToken(string): session token
+    """
+    role_id = g.user.role_id
+    if not role_id:
+        raise ApiException(requests.codes.bad_request,
+                           "You have no attached role to assume")
+    role = Role.query.filter_by(id=role_id).one_or_none()
+    if not role:
```

Comment:
Shouldn't be possible - i.e., this should be enforced by Postgres.

You can also add a `db.relationship` to the `User` class so you can just access `g.user.role` instead of running a query manually. (There are examples in other classes in model.py)

Reply:
Still not possible?

Reply:
removed